src/relayserver/RelayServer.ts

```diff
@@ -624,18 +624,21 @@ latestBlock timestamp   | ${latestBlock.timestamp}
     return events
   }
 
-  async _handleTransactionRelayedEvent (paymaster: Address, blockNumber: number): Promise<void> {
+  async _handleTransactionRelayedEvent (paymaster: Address, eventBlockNumber: number): Promise<void> {
     if (this.config.runPaymasterReputations) {
-      await this.reputationManager.updatePaymasterStatus(paymaster, true)
+      await this.reputationManager.updatePaymasterStatus(paymaster, true, eventBlockNumber)
     }
   }
 
-  async _handleTransactionRejectedByPaymasterEvent (paymaster: Address, blockNumber: number): Promise<void> {
-    this.alerted = true
-    this.alertedBlock = blockNumber
-    this.logger.error(`Relay entered alerted state. Block number: ${blockNumber}`)
+  async _handleTransactionRejectedByPaymasterEvent (paymaster: Address, currentBlockNumber: number, eventBlockNumber: number): Promise<void> {
+    // TODO: do not call this method when events are processed already (stateful server thing)
```

Comment:
comment irrelevant.

Reply:
I'll keep it because I want to stop calling 'handes' for past events